corehq/apps/user_importer/importer.py

```diff
@@ -494,18 +493,23 @@ def _get_domain_info(domain):
                         if current_user and not current_user.is_member_of(domain) and is_account_confirmed:
                             current_user.add_as_web_user(domain, role=role_qualified_id, location_id=user.location_id)
                         elif not current_user or not current_user.is_member_of(domain):
-                            invite_data = {
-                                'email': web_user,
-                                'invited_by': upload_user.user_id,
-                                'invited_on': datetime.utcnow(),
-                                'domain': domain,
-                                'role': role_qualified_id,
-                                'supply_point': user.location_id
-                            }
-                            invite = Invitation(**invite_data)
-                            invite.save()
-                            if send_account_confirmation_email:
-                                invite.send_activation_email()
+                            if len(Invitation.by_email(web_user)) >= 1:  # an invitation already exists
```

Comment:
This would be a good place to use [update_or_create](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#update-or-create)